src/hw_ostc_parser.c

```diff
@@ -1049,6 +1080,10 @@ hw_ostc_parser_samples_foreach (dc_parser_t *abstract, dc_sample_callback_t call
 				return DC_STATUS_DATAFORMAT;
 			}
 			unsigned int idx = data[offset];
+			if (version == 0x24 && data[layout->divemode] == OSTC3_CC) {
+				// OSTC4 reports gas changes to another diluent with an offset
+				idx -= CC_DILUENT_GAS_OFFSET;
```

Comment:
Is this offset actually guaranteed to be there? What if this isn't a diluent gas at all?

IOW, maybe this all shouldn't be conditional on "version 0x24 and CC", but on 'idx' having a large value?

Reply:
From looking at the [source](https://code.heinrichsweikamp.com/public/ostc4/file/tip) for the firmware currently released for the OSTC4, this offset is applied on the OSTC4 _for every dive in CCR mode_. The offset is _not_ applied on the OSTC4 when it is in OC mode - so this check is definitely needed.
As for the "version is `0x24` check, the background for this is that both OSTC4 and a  number of other Heinrichs Weikamp dive computers are both using the 'ostc3' 'dialect' of the hw_ostc protocol. The differentiation between the 'sub-dialects' is made by the `version` byte, with `0x24` standing for OSTC4. (I think using defines for `0x23` and `0x24` would greatly simplify reading this code.)
These other dive computers use a [completely different firmware](https://code.heinrichsweikamp.com/public/hwos_code/file/tip/src), and from talking to @jefdriesen this firmware does not have this offset for gas changes - I cannot test this.
Also from looking at the OSTC4 source, only changes to a diluent gas are reported through a `0x20` gaschange event - changes to an OC bailout gas are always reported as a `0x0100` bailout event.
As for generally applying an offset if the reported index is in between 5 and 10 I am confident that this would work for the current firmware for OSTC4, but not sure if this will work for the other firmware using the same code path - there might be cases where it legitimately returns gas mixes in this range.

Also note that I've pushed another update - I realised that during the rebase I had accidentally removed the code pushing the gas use type back into Subsurface.